python/ray/tune/suggest/suggestion.py

```diff
@@ -212,6 +216,63 @@ def add_evaluated_point(self,
         """
         raise NotImplementedError
 
+    def add_evaluated_trials(self, trials_or_analysis: Union["Trial", List[
+            "Trial"], "ExperimentAnalysis"], metric: str):
+        """Pass results from trials that have been evaluated separately.
+
+        This method allows for information from outside the
+        suggest - on_trial_complete loop to be passed to the search
+        algorithm.
+        This functionality depends on the underlying search algorithm
+        and may not be always available (same as ``add_evaluated_point``.)
+
+        Args:
+            trials_or_analysis (Trial|List[Trial]|ExperimentAnalysis):
+                Trials to pass results form to the searcher.
+            metric (str): Metric name reported by trials used for
+                determining the objective value.
+
+        """
+        if self.add_evaluated_point == Searcher.add_evaluated_point:
+            raise NotImplementedError
+
+        # lazy imports to avoid circular dependencies
+        from ray.tune.trial import Trial
+        from ray.tune.analysis import ExperimentAnalysis
+        from ray.tune.result import DONE
+
+        if isinstance(trials_or_analysis, Trial):
+            trials_or_analysis = [trials_or_analysis]
+        elif isinstance(trials_or_analysis, ExperimentAnalysis):
+            trials_or_analysis = trials_or_analysis.trials
+
+        any_trial_had_metric = False
+
+        def trial_to_points(trial: Trial) -> Dict[str, Any]:
+            nonlocal any_trial_had_metric
+            has_trial_been_pruned = (trial.status == Trial.TERMINATED and
+                                     not trial.last_result.get(DONE, False))
+            has_trial_finished = (trial.status == trial.TERMINATED
+                                  and trial.last_result.get(DONE, False))
+            if not any_trial_had_metric:
+                any_trial_had_metric = (metric in trial.last_result
+                                        and has_trial_finished)
+            return dict(
+                parameters=trial.config,
+                value=trial.last_result.get(metric, None),
+                error=trial.status == Trial.ERROR
+                or metric not in trial.last_result,
```

Comment:
Is this really an error then? Should we maybe just throw a warning for these trials and not report them?

Reply:
Will now skip the trial